src/KubernetesClient/WatcherDelegatingHandler.cs

```diff
@@ -107,24 +172,18 @@ protected override bool TryComputeLength(out long length)
             }
         }
 
-        internal class PeekableStreamReader : StreamReader
+        internal class PeekableStreamReader : TextReader
         {
-            private Queue<string> _buffer;
+            private readonly Queue<string> _buffer;
+            private readonly StreamReader _inner;
 
-            public PeekableStreamReader(Stream stream) : base(stream)
+            public PeekableStreamReader(Stream stream)
             {
                 _buffer = new Queue<string>();
+                _inner = new StreamReader(stream);
             }
 
-            public override string ReadLine()
-            {
-                if (_buffer.Count > 0)
-                {
-                    return _buffer.Dequeue();
-                }
-
-                return base.ReadLine();
-            }
+            public override string ReadLine() => throw new NotImplementedException();
```

Comment:
Shouldn't we leave this implementation?

Reply:
The method is effectively unused now. It was made to throw for consistency with the other unused members.